fnet/network.go

```diff
@@ -111,7 +135,8 @@ func ResolveDestination(dest string) *net.TCPAddr {
 
 // Resolve returns the TCP address of the host,port suitable for net.Dial.
 // nil in case of errors.
-func Resolve(host string, port string) *net.TCPAddr {
+func Resolve(host string, port string) net.Addr {
```

Comment:
Do we really need to change here to `net.Addr` because if I'm not thinking wrong, we are using `resolve` method for TCP ADDR. It might return again '*net.TCPAddr'

Reply:
You are right, I made that change when I was going down the path of trying to hack into the host/port part of the url some kind of scheme to parse UDS. golang's url parse rejects encoding inside the host/part so I gave up and went for UnixDomainSocket option instead

This being said it doesn't hurt to have the option to return other type of addresses and most of the code now does take the generic Addr instead of TCPAddr